pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -1099,29 +1102,8 @@ func newClusterRoleBinding(ns *corev1.Namespace) *rbacv1.ClusterRoleBinding {
 	}
 }
 
-// newPod creates a pod with 2 containers with a shared volume mount:
-// - gather: init containers that run gather command
-// - copy: no-op container we can exec into
-func (o *MustGatherOptions) newPod(node, image string, hasMaster bool, affinity *corev1.Affinity) *corev1.Pod {
-
+func defaultMustGatherPod(image string, update func(pod *corev1.Pod)) *corev1.Pod {
```

Comment:
I would be happier for this function not to contain `update` arg. You can just do the same with

```
pod := defaultMustGatherPod()
pod.XYZ = ...
```

I see it's handy for tests, though, so feel free to define a test helper that lets you pass a callback.